lib/Parse/ParsePattern.cpp

```diff
@@ -67,10 +67,10 @@ void Parser::DefaultArgumentInfo::setFunctionContext(
 static ParserStatus parseDefaultArgument(
     Parser &P, Parser::DefaultArgumentInfo *defaultArgs, unsigned argIndex,
     Expr *&init, bool &hasInheritedDefaultArg,
-    Parser::ParameterContextKind paramContext) {
+    Parser::ParameterContextKind paramContext, tok assignmentTok) {
   SyntaxParsingContext DefaultArgContext(P.SyntaxContext,
                                          SyntaxKind::InitializerClause);
-  SourceLoc equalLoc = P.consumeToken(tok::equal);
+  SourceLoc equalLoc = P.consumeToken(assignmentTok);
```

Comment:
I wonder if consumption of either '=' or '==' should be taken out of here and be performed on caller site?

Reply:
`consumeToken` also allows not passing a token kind at all, but I still think you might be right. It's weird to consume something when you don't know what it is.

Reply:
I'm OK with:
```cpp
assert(Tok.is(tok::equal) ||
       (Tok.isBinaryOperator() && Tok.getText() == "=="));
consumeToken();
```
like
https://github.com/apple/swift/blob/26c4cccb4b971abae950d6a183354f70b2e3c0c7/lib/Parse/ParseDecl.cpp#L6632-L6634

Reply:
thanks for the feedback .. address in https://github.com/apple/swift/pull/26139/commits/2cfe8ad52a0f2162808381101b6b80cb75a2b118